timeseries/src/autogluon/timeseries/models/ensemble/abstract.py

```diff
@@ -77,10 +83,48 @@ def remap_base_models(self, model_refit_map: Dict[str, str]) -> None:
         """
         raise NotImplementedError
 
-    # TODO: remove
-    def _fit(*args, **kwargs):
-        pass
 
-    # TODO: remove
-    def _predict(*args, **kwargs):
-        pass
+class AbstractWeightedTimeSeriesEnsembleModel(AbstractTimeSeriesEnsembleModel):
+    """Abstract class for weighted ensembles which assign one (global) weight per model."""
+
+    def __init__(self, name: Optional[str] = None, **kwargs):
+        if name is None:
+            name = "WeightedEnsemble"
+        super().__init__(name=name, **kwargs)
+        self.model_to_weight: Dict[str, float] = {}
+
+    @property
+    def model_names(self) -> List[str]:
+        return list(self.model_to_weight.keys())
+
+    @property
+    def model_weights(self) -> np.ndarray:
+        return np.array(list(self.model_to_weight.values()), dtype=np.float64)
+
+    def predict(self, data: Dict[str, TimeSeriesDataFrame], **kwargs) -> TimeSeriesDataFrame:
+        if not set(self.model_names).issubset(set(data.keys())):
+            raise ValueError(
+                f"Set of models given for prediction in {self.name} differ from those provided during initialization."
+            )
+        for model_name, model_pred in data.items():
+            if model_pred is None:
+                raise RuntimeError(f"{self.name} cannot predict because base model {model_name} failed.")
+
+        # Make sure that all predictions have same shape
+        assert len(set(pred.shape for pred in data.values())) == 1
+
+        weighted_predictions = [data[model_name] * weight for model_name, weight in self.model_to_weight.items()]
+
+        return functools.reduce(lambda x, y: x + y, weighted_predictions)
+
+    def get_info(self) -> dict:
+        info = super().get_info()
+        info["model_weights"] = self.model_to_weight
```

Comment:
Let's return a copy to make sure that the user doesn't accidentally break the ensemble by fiddling with the info?